### DISCUSSION TITLE
List of frameworks with complex file structures

### DISCUSSION THREAD

**Discussion by lukaspetersson** [AntonOsika/gpt-engineer#707] [Ideas]
Many frameworks, like ReactJS, have intricate file structures that are often challenging for the gpt-engineer to accurately generate. This can be frustrating since it results in non-executable code.

Addressing this through prompt engineering could be a solution. However, a more straightforward approach might be to hardcode the file structures for the most popular frameworks.

This aligns with the "Simple to get value" philosophy, though it might not align with the long-term vision of the project.

What are your thoughts? Which file structures would you like to see supported?

**Comment by ATheorell** [comment]
Personal opinion here: Sounds like smart engineering! I think we should aim to get the common problems to work first, rather than directly aiming for a general solution. No clear idea exactly how it would be enforced or what specific frameworks to support.

**Reply by lukaspetersson** [reply]
For me reactJS would be nice, then we could hardcode in commands to generate an empty project, install packages and start dev server.
